### ISSUE TITLE
Add config file path getter.

### ISSUE THREAD

**Issue by ykmnkmi** [dart-lang/package_config#126]
```dart
import 'dart:io' show File;

import 'package:package_config/package_config.dart';

// fire ./app/bin/main.dart
Future<void> main(List<String> arguments) async {
  var filePath = arguments[0];
  var file = File(filePath);

  var config = await findPackageConfig(file.parent);
  String? configPath;

  if (config != null) {
    // getter like this
    configPath = config.relativePath;
  }

  // ./app/.dart_tool/package_config.json
  print(configPath);
}
```

**Comment by dcharkes** [comment]
This would enable us to unify the logic between this package and the front_end. Currently, the frontend has its own logic for recursively traversing the folder structure upwards.

pkg/front_end/lib/src/base/processed_options.dart

```dart
  /// Get the package map which maps package names to URIs.
  ///
  /// This is an asynchronous getter since file system operations may be
  /// required to locate/read the packages file.
  Future<PackageConfig> _getPackages() async {
```

cc @sigmundch, @jensjoha

**Comment by jensjoha** [comment]
> This would enable us to unify the logic between this package and the front_end. Currently, the frontend has its own logic for recursively traversing the folder structure upwards.

It would probably not. `package:package_config` works on `Directory` etc (i.e. the local filesystem) --- the CFE has it's own file system abstraction (e.g. for working with build systems).

**Comment by dcharkes** [comment]
And this would enable us to reuse the logic in dartdev.

https://dart-review.googlesource.com/c/sdk/+/267340/74/pkg/dartdev/lib/src/commands/build.dart#194

Alternatively, the logic for finding the package config file should be exposed separately. In case we want to find the package config but not necessarily read it.

**Comment by mkustermann** [comment]
@lrhn Could we add an API to `package:package_config` to return the path to the `.dart_tool/package_config.json` file?

**Comment by lrhn** [comment]
We could add a function which returns the path (`File`) of a package config file, without actually reading it. The code is probably already there, used to find the file before parsing it 

That code would probably not look for `.packages` file. Or maybe it could, if you opt in to it, since the file returned includes the file name.

**Comment by dcharkes** [comment]
> We could add a function which returns the path (File) of a package config file, without actually reading it. The code is probably already there, used to find the file before parsing it

No, it's not. While recursing up the directory structure we parse candidates and check the `minVersion`, so we do actually need to read the contents of the file to have these semantics.

Everything else in the eco-system does not read the file before deciding (the VM, the frontend etc.), so maybe we should consider doing a breaking change to `package:package_config` so that we can do as you propose.

> That code would probably not look for .packages file. Or maybe it could, if you opt in to it, since the file returned includes the file name.

The `minVersion` argument would control if we consider that file yes or no.

For my purpose it doesn't really matter whether it's a separate function that can be used to find the file, or whether we add a `path` getter to `PackageConfig` itself. In many places in dartdev/flutter_tools and the support packages dealing with native assets I often need both. (I guess that's an argument for adding the path getter instead, so that we only have to traverse the directory structure once.)

If we go for a function that returns the path, we would also have the `recurse` argument, so that this can also be used by callers which do not want to recurse up the directory structure.

**Comment by kenzieschmoll** [comment]
+1 to the request for adding an API to get the package config path. We need this so that we can lookup the package config URI from the PackageConfig object in flutter tools: https://github.com/flutter/flutter/pull/150440
